cmd/containerboot/settings.go

```diff
@@ -98,6 +99,7 @@ func configFromEnv() (*settings, error) {
 		PodIP:                                 defaultEnv("POD_IP", ""),
 		EnableForwardingOptimizations:         defaultBool("TS_EXPERIMENTAL_ENABLE_FORWARDING_OPTIMIZATIONS", false),
 		HealthCheckAddrPort:                   defaultEnv("TS_HEALTHCHECK_ADDR_PORT", ""),
+		UserMetricsAddrPort:                   defaultEnv("TS_USERMETRICS_ADDR_PORT", ""),
```

Comment:
What I am concerned about is that for container users we will now have two very ways how to expose different sets of metrics, one via this env var, another one via conffile/debug flag.

I would like to understand if we expect people who currently consume the internal client metrics to stop doing that and just use these metrics (which in that case would provide all the same metrics that they might need)? Or, if we expect that client metrics will still be useful for folks in future, can we make it so that both sets are enabled via the same mechanism? Or, do we expect that there will be users who will want to use just one and not the other (performance etc)?

Generally, not a problem to be solved in this PR, but I would like us to have an understanding of what we want this to look like in future before adding a new env var. Or, we could add it with `EXPERIMENTAL_` prefix .

Reply:
Similar with `ProxyClass`, I'd like to understand if we should aim for a default being that only the new metrics are enabled. If so, perhaps we should add a new `proxyClass.spec.metrics.debugMetrics` field and default that to `off` over a few releases?

Reply:
I don't think we expect users to consume any of the other two sets of metrics [exposed](https://github.com/tailscale/tailscale/blob/b9ecc50ce38d23cfacd5fae6360fa9742c0564a6/cmd/tailscaled/tailscaled.go#L776-L777) by the tailscaled debug web server (`expvar` and `clientmetric` metrics). Those metrics are not documented, not tested, and are sometimes wrong or misleading. They might be helpful on a case by case basis though - I believe we've occasionally asked customers to enable these as part of an ongoing investigation (via support), and they might be useful for other OSS developers that use Tailscale client. But I think it would be a mistake for anyone to use these on an ongoing basis.

The new set of metrics have been added specifically for users, and will be documented in the KB. They are the metrics we'd suggest folks to put on their Grafana dashboards. It's a much smaller set of metrics, but the one we are committing to support longer term.

With this in mind, my recommendation would be to not have both sets of metrics enabled via the same mechanism. Ideally I think we'd want the user-facing metrics to be enabled via a well documented knob called "metrics", while the internal ones should probably require a more generic "debug" knob (like the `--debug` flag in `tailscaled`).

Reply:
Thanks, so the TL;DR is that we want users to only use the new metrics and those will aim to satisfy all regular monitoring needs.

@tomhjp I think in that case we want to aim to make it so that over a couple releases `proxyClass.spec.metrics.enabled` only enables the new metrics endpoint and there is another knob for the debug endpoint that could start off by being set to `true` for now with warnings.